### DISCUSSION TITLE
Metamask - Same address for all networks

### DISCUSSION THREAD

**Discussion by programonauta** [smartcontractkit/full-blockchain-solidity-course-py#532] [Q&A] [answered]
Hey guys,

Just a question for curiosity,

I noticed that Metamask uses the same address for different networks (Ethereum Mainnet, Binance, Testnets, and so on).

Is there any reason for that?

**Comment by cromewar** [comment, marked as answer]
The reason is related to the private key, as you have the same private key the algorithm with generate a public key using that password, so it will be the same for all chains using that private key.

If you want to know more about how encryption works, I suggest you to check [this](https://www.preveil.com/blog/public-and-private-key/) link.

Best regards Cromewar.

**Reply by programonauta** [reply]
Thanks @cromewar
